File: src/services/sms-api.ts
export class SmsService {
  private baseURL = "https://tethys-sms.onrender.com";

  constructor() {}

  async sendSms(phoneNumber: string, message: string) {
    const url = `${this.baseURL}/enviar-dados`;
    const localizedPhoneNumber = phoneNumber.includes("+") ? phoneNumber : `+55${phoneNumber}`

    console.log(`URL: ${url}`)
    console.log(`Enviando SMS para ${localizedPhoneNumber}`)
    console.log(`Mensagem: ${message}`)

    const response = await fetch(url, {
      method: "POST",
      headers: {
        "Content-Type": "application/x-www-form-urlencoded",
      },
      body: new URLSearchParams({
        numero: localizedPhoneNumber,
        texto: message,
      }),
    });

    console.log(await response.json())

    return response;
  }
}

File: services/sms_api.py
import requests

class SmsService:
    base_url = "https://tethys-sms.onrender.com"

    def __init__(self):
        self.url = self.base_url + "/enviar-dados"

    def send(self, phone: str, message: str):
        print(f"Enviando SMS para {phone}")

        phone = ''.join(filter(str.isdigit, phone))
        if not phone.startswith('+'): phone = '+55' + phone

        headers = { "Content-Type":  "application/x-www-form-urlencoded" }
        data = { "numero": phone, "texto": message }
        response = requests.post(self.url, headers=headers, data=data)
        return response.json()
